activerecord/test/cases/inheritance_test.rb

```diff
@@ -227,6 +229,14 @@ def test_alt_becomes_works_with_sti
     assert_kind_of Cabbage, cabbage
   end
 
+  def test_becomes_sets_variables_before_initialization_callbacks
+    category = Category.find(1)
+    assert_kind_of Category, category
+    special_category = category.becomes(SpecialCategory)
+    assert_equal "GENE", special_category.special_name, "General special_name should be GENE"
+    assert_equal "GENE", category.special_name, "General special_name should be GENE"
```

Comment:
Could this test use the Vegetable model instead? It's already used in this file, and isn't used elsewhere, while Category is used throughout the test suite.

Reply:
@eugenius probably. I picked SpecialCategory because there aren't any tests that involve checking the attributes in SpecialCategory, whereas all of the other inheritance models did; and adding a modified attribute in Vegetable seemed more likely to unintentionally affect other unrelated tests

Reply:
updated to use Vegetable and a new `YellingVegetable` submodel